### DISCUSSION TITLE
Customizing and extending the Formatting Toolbar

### DISCUSSION THREAD

**Discussion by magdalenapaciorek** [WordPress/developer-blog-content#255] [Topic Ideas]
I'm proposing this topic (and volunteering myself as the author) because patterns similar to the one below are quite common in web design. I believe the formatting API is often overlooked, and developers who are unfamiliar with it might struggle to implement such designs.

![block-formatting-toolbar](https://github.com/WordPress/developer-blog-content/assets/13915987/470cc434-5435-488a-bda1-63f0b70f8775)

The initial plan for the article:

- an overview of available formats with examples of how they can be used (e.g., adding an inline image to the citation in the Quote Block)
- how to unregister default formats
- how to add a basic format, similar to the one shown in the screenshot
- how to implement more complex format controls, such as the one in the Language Format, where a click opens a popover with multiple options.

**Comment by bph** [comment]
Topic approved 5/2/24 the conversation continues on the issue
- #261
